pkg/reconciler/broker/trigger/trigger.go

```diff
@@ -194,10 +195,7 @@ func (r *Reconciler) subscribeToBrokerChannel(ctx context.Context, b *eventingv1
 		Namespace:  b.Namespace,
 	}
 
-	delivery := t.Spec.Delivery
-	if delivery == nil {
-		delivery = b.Spec.Delivery
-	}
```

Comment:
I agree that the current solution is simplisttic 

I think we should ensure the behavior being aligned on other broker implementations

/cc @xtreme-sameer-vohra